File: app/src/main/java/com/nruge/iceinfo/model/FacilityModels.kt
package com.nruge.iceinfo.model

enum class FacilityType {
    ELEVATOR, ESCALATOR, TOILET, WIFI, INFO_DESK, DEPARTURE_MONITOR,
    RAMP, PARKING, BIKE_PARKING, WAITING_ROOM
}

enum class FacilityStatus { ACTIVE, INACTIVE, UNKNOWN }

data class StationFacility(
    val id: String,
    val type: FacilityType,
    val label: String,
    val status: FacilityStatus,
    val description: String = ""
)

data class StationInfo(
    val evaNr: String,
    val name: String,
    val liveFacilities: List<StationFacility> = emptyList(),
    val staticFacilities: List<FacilityType> = emptyList(),
    val isLoading: Boolean = false,
    val error: String? = null
)

data class StationSearchResult(
    val evaNr: String,
    val name: String
)

File: ios/ICEInfo/Models/FacilityModels.swift
import Foundation

enum FacilityType: String, Codable, Sendable {
    case elevator = "ELEVATOR"
    case escalator = "ESCALATOR"
    case toilet = "TOILET"
    case wifi = "WIFI"
    case infoDesk = "INFO_DESK"
    case departureMonitor = "DEPARTURE_MONITOR"
    case ramp = "RAMP"
    case parking = "PARKING"
    case bikeParking = "BIKE_PARKING"
    case waitingRoom = "WAITING_ROOM"
}

enum FacilityStatus: String, Codable, Sendable {
    case active = "ACTIVE"
    case inactive = "INACTIVE"
    case unknown = "UNKNOWN"
}

struct StationFacility: Codable, Sendable, Identifiable {
    var id: String
    var type: FacilityType
    var label: String
    var status: FacilityStatus
    var description: String = ""
}

struct StationInfo: Codable, Sendable {
    var evaNr: String
    var name: String
    var liveFacilities: [StationFacility] = []
    var staticFacilities: [FacilityType] = []
    var isLoading: Bool = false
    var error: String? = nil
}

struct StationSearchResult: Codable, Sendable, Identifiable {
    var id: String { evaNr }
    var evaNr: String
    var name: String
}
